builtin/logical/database/mockv4.go

```diff
@@ -92,7 +92,7 @@ func (m MockDatabaseV4) RotateRootCredentials(ctx context.Context, statements []
 	if err != nil {
 		return config, fmt.Errorf("failed to generate credentials: %w", err)
 	}
-	config["password"] = newPassword
+	m.config["password"] = newPassword
```

Comment:
`config` is always nil at this point, so I assume the intent was to assign the password to `m.config` instead.